Content.Server/GameObjects/Components/Body/Behavior/BrainBehaviorComponent.cs

```diff
@@ -0,0 +1,91 @@
+﻿#nullable enable
+using Content.Server.GameObjects.Components.Mobs;
+using Content.Server.Mobs;
+using Content.Shared.GameObjects.Components.Body;
+using Content.Shared.GameObjects.Components.Body.Behavior;
+using Content.Shared.GameObjects.Components.Body.Part;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.ComponentDependencies;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Log;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Body.Behavior
+{
+    [RegisterComponent]
+    public class BrainBehaviorComponent : MechanismBehaviorComponent
+    {
+        public override string Name => "Brain";
+
+        protected override void OnAddedToBody()
+        {
+            base.OnAddedToBody();
+
+            if (Body == null)
+                return;
+
+            HandleMind(Body.Owner, Owner);
+        }
+
+        protected override void OnAddedToPart()
+        {
+            base.OnAddedToPart();
+
+            if (Part == null)
+                return;
+
+            HandleMind(Part.Owner, Owner);
+        }
+
+        protected override void OnAddedToPartInBody()
+        {
+            base.OnAddedToPartInBody();
+
+            if (Body == null || Part == null)
+                return;
+
+            HandleMind(Body.Owner, Owner);
+        }
+
+        protected override void OnRemovedFromBody(IBody old)
+        {
+            base.OnRemovedFromBody(old);
+
+            if (Part != null)
+            {
+                HandleMind(Part.Owner, old.Owner);
+                return;
+            }
+
+            HandleMind(Owner, old.Owner);
+        }
+
+        protected override void OnRemovedFromPart(IBodyPart old)
+        {
+            base.OnRemovedFromPart(old);
+
+            HandleMind(Owner, old.Owner);
+        }
+
+        protected override void OnRemovedFromPartInBody(IBody? oldBody, IBodyPart? oldPart)
+        {
+            base.OnRemovedFromPartInBody(oldBody, oldPart);
+
+            if (oldBody == null) return;
+
+            HandleMind(oldBody.Owner, Owner);
+        }
```

Comment:
I made a mistake here with the nullability, oldBody and oldPart should never be null. This is a leftover from how it was done before.